framework/helpers/BaseStringHelper.php

```diff
@@ -141,6 +141,41 @@ public static function truncateWords($string, $count, $suffix = '...', $asHtml =
 
         return $string;
     }
+    
+    
+    /**
+     * Truncates a string in the middle. Keeping start and end.
+     * `StringHelper::truncateMiddle('Hello world number 2', 8)` produces "Hell...er 2".
+     *
+     * This method does not support html. It will strip all tags even if length is smaller.
+     *
+     * @param string $string    The string to truncate.
+     * @param int $length       How many characters from original string to include into truncated string.
+     * @param string $separator String to append in the middle of truncated string.
+     * @param string $encoding  The charset to use, defaults to charset currently used by application.
+     * @return string the truncated string.
+     */
+    public static function truncateMiddle($string, $length, $separator = '...', $encoding = null)
+    {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
+        }
+
+        $string = strip_tags($string);
```

Comment:
This doesn't seem to be right. Should work with HTML as well.

Reply:
I cant find a way to do this without breaking BC, because method `truncateHTML` does not use a flag to determine wether it should be truncated by words or by characters. Instead it uses `$encoding` value. If it is false, then words, if is null or other value it truncate by characters.

Reply:
Then it's probably a good idea to introduce a separate method for HTML. Don't do it now though, we should verify that the feature is useful to more people than you and, probably, me.

Reply:
Ok.